src/driver/amdxdna/amdxdna_of_drv.c

```diff
@@ -71,6 +73,17 @@ static int amdxdna_of_probe(struct platform_device *pdev)
 		XDNA_WARN(xdna, "DMA configuration downgraded to 32bit Mask\n");
 	}
 
+	xrs_cfg.ddev = &xdna->ddev;
+	xrs_cfg.total_col = XRS_MAX_COL;
+
+	if (xdna->dev_handle)
```

Comment:
xdna->dev_handle
This one checks twice.
```suggestion
	        if (xdna->dev_handle) {
                xdna->dev_handle->xrs_hdl = xrsm_init(&xrs_cfg);
                if (!xdna->dev_handle->xrs_hdl) {
                        XDNA_ERR(xdna, "Initialize resolver failed");
                        drm_dev_put(&xdna->ddev);
                        return -EINVAL;
                }
        }

```